components/KeyboardAvoidingLayout.js

```diff
@@ -0,0 +1,45 @@
+const T = require('prop-types');
+const { KeyboardAvoidingView, Platform, ScrollView } = require('react-native');
+const { useHeaderHeight } = require('@react-navigation/stack');
```

Comment:
In the next major release of `react-navigation` (`6.x`), this hook will be found in [@react-navigation/elements](https://reactnavigation.org/docs/elements/#useheaderheight)